src/Internal/Service/Blockfrost.purs

```diff
@@ -253,11 +379,360 @@ getTxMetadata txHash = do
+    :: Int -> BlockfrostServiceM (Either ClientError BlockfrostUtxosAtAddress)
+  utxosAtAddressOnPage page = runExceptT do
+    -- Maximum number of results per page supported by Blockfrost:
+    let maxNumResultsOnPage = 100
+    utxos <- ExceptT $ handleBlockfrostResponse <$>
+      blockfrostGetRequest (UtxosAtAddress address page maxNumResultsOnPage)
+    case Array.length (unwrap utxos) < maxNumResultsOnPage of
+      true -> pure utxos
+      false -> append utxos <$> ExceptT (utxosAtAddressOnPage $ page + 1)
+
+getUtxoByOref
+  :: TransactionInput
+  -> BlockfrostServiceM (Either ClientError (Maybe TransactionOutput))
+getUtxoByOref oref@(TransactionInput { transactionId: txHash }) = runExceptT do
+  (blockfrostUtxoMap :: BlockfrostUtxosOfTransaction) <-
+    ExceptT $ handleBlockfrostResponse <$>
+      blockfrostGetRequest (UtxosOfTransaction txHash)
+  traverse (ExceptT <<< resolveBlockfrostTxOutput)
+    (snd <$> Array.find (eq oref <<< fst) (unwrap blockfrostUtxoMap))
+
+--------------------------------------------------------------------------------
+-- Get datum by hash
+--------------------------------------------------------------------------------
+
+getDatumByHash
+  :: DataHash -> BlockfrostServiceM (Either ClientError (Maybe Datum))
+getDatumByHash dataHash = do
+  response <- blockfrostGetRequest (DatumCbor dataHash)
+  pure $ handle404AsNothing $ unwrapBlockfrostDatum <$> handleBlockfrostResponse
+    response
+
+--------------------------------------------------------------------------------
+-- Get script by hash
+--------------------------------------------------------------------------------
+
+getScriptByHash
+  :: ScriptHash
+  -> BlockfrostServiceM (Either ClientError (Maybe ScriptRef))
+getScriptByHash scriptHash = runExceptT $ runMaybeT do
+  scriptInfo <- MaybeT $ ExceptT $ getScriptInfo scriptHash
+  case scriptLanguage scriptInfo of
+    NativeScript ->
+      NativeScriptRef <$>
+        (MaybeT $ ExceptT getNativeScriptByHash)
+    PlutusV1Script ->
+      PlutusScriptRef <$> plutusV1Script <$>
+        (MaybeT $ ExceptT getPlutusScriptCborByHash)
+    PlutusV2Script ->
+      PlutusScriptRef <$> plutusV2Script <$>
+        (MaybeT $ ExceptT getPlutusScriptCborByHash)
+  where
+  getNativeScriptByHash
+    :: BlockfrostServiceM (Either ClientError (Maybe NativeScript))
+  getNativeScriptByHash = runExceptT do
+    (nativeScript :: Maybe BlockfrostNativeScript) <- ExceptT do
+      response <- blockfrostGetRequest (NativeScriptByHash scriptHash)
+      pure $ handle404AsNothing $ handleBlockfrostResponse response
+    pure $ unwrap <$> nativeScript
+
+  getPlutusScriptCborByHash
+    :: BlockfrostServiceM (Either ClientError (Maybe ByteArray))
+  getPlutusScriptCborByHash = runExceptT do
+    (plutusScriptCbor :: Maybe BlockfrostCbor) <- ExceptT do
+      response <- blockfrostGetRequest (PlutusScriptCborByHash scriptHash)
+      pure $ handle404AsNothing $ handleBlockfrostResponse response
```

Comment:
There's some redundant null checking here, I think. handle404AsNothing causes a null check by forcing handleBlockfrostResponse to output a `Maybe`, which will do a null check. Then `BlockfrostCbor` does a null check by internally wrapping a `Maybe` too.

Not the end of the world, just wondering if there's an easier way? Unwrap `BlockfrostCbor` before being passed into `handle404AsNothing`?